### PR TITLE
Create an 'extension' subcommand

### LINKED ISSUE #239: Make nbgrader install a real subcommand
Now that the install command is a proper app, we might as well make it a proper subcommand of nbgrader so users can see it in the command line documentation and run it like the other commands.

### PR DESCRIPTION
**Pull request by jhamrick**
Fixes #239 

I created a new subcommand called `extension`, and gave it three different subcommands as well:

* `nbgrader extension install`
* `nbgrader extension activate`
* `nbgrader extension deactivate`

Note that this does break existing syntax. I think it's a little nicer this way, though -- it makes it clearer that the installation is specific to *only* the nbextension, and not nbgrader in general, and separates installation from activation, which I think can be a little confusing.

Ping @ellisonbg @Carreau

### CODE REVIEW

**Comment by jhamrick** [comment]
Ping @Carreau again, as requested :)

File: nbgrader/apps/extensionapp.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+    def start(self):
+        self.extra_args = [os.path.abspath(os.path.join(os.path.dirname(__file__), '..', 'nbextensions', 'nbgrader'))]
+        self.install_extensions()
+
+
+activate_flags = {}
+activate_flags.update(base_flags)
+activate_flags.update({
+})
```
Comment (Carreau) [inline nbgrader/apps/extensionapp.py]: extend with empty dict.

File: nbgrader/apps/extensionapp.py [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+            os.mkdir(json_dir)
+
+        with io.open(json_file, 'w+') as f:
+            f.write(cast_unicode_py2(json.dumps(config, indent=2), 'utf-8'))
+
+
+deactivate_flags = {}
+deactivate_flags.update(base_flags)
+deactivate_flags.update({
+})
```
Comment (Carreau) [inline nbgrader/apps/extensionapp.py]: empty again.

**Comment by Carreau** [comment]
I had a quick look, I'll try to see if we can abstract that into a common package from which you can inherit and just set default value, to have a common user interface across extensions.

I came across [this](http://click.pocoo.org/4/) today also to build CLI interfaces, I should have a look. 

I'll re-take a look tomorrow.

**Comment by jhamrick** [comment]
Thanks @Carreau ! I think I will go ahead and merge this and then if some common package emerges for activating extensions I can update this again later... currently nbgrader still depends on 3.x and I imagine that change would be for 4.0, so I think this is a thing we can revisit after 4.0 is released and I update nbgrader to make it compatible with the split repos.

**Pull request merged**